bin/improver-temporal-interpolate

```diff
@@ -0,0 +1,121 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Script to interpolate data between validity times"""
+
+import warnings
+
+from improver.argparser import ArgParser
+from improver.utilities.load import load_cube
+from improver.utilities.save import save_netcdf
+from improver.utilities.temporal import (
+    cycletime_to_datetime, TemporalInterpolation)
+
+
+def main():
+    """
+    Interpolate data to intermediate times between the validity times of two
+    cubes. This can be used to fill in missing data (e.g. for radar fields) or
+    to ensure data is available at the required intervals when model data is
+    not available at these times.
+    """
+    parser = ArgParser(
+        description='Interpolate data between validity times ')
+
+    parser.add_argument('data_t0', metavar='DATA_T0',
+                        help='Cube contain the data at the beginning'
+                        ' of the period')
+
+    parser.add_argument('data_t1', metavar='DATA_T1',
+                        help='Cube contain the data at the end'
+                        ' of the period')
+
+    group = parser.add_mutually_exclusive_group(required=True)
+
+    group.add_argument("--interval_in_mins", metavar="INTERVAL_IN_MINS",
+                       default=None, type=int,
+                       help="Specifies the interval in minutes"
+                       " at which to interpolate "
+                       "between the two input cubes."
+                       " A number of minutes which does not "
+                       "divide up the interval equally will "
+                       "raise an exception. If intervals_in_mins "
+                       "is set then times can not be set.")
+
+    group.add_argument("--times", metavar="TIMES",
+                       default=None, nargs="+", type=str,
+                       help="Specifies the times in the format "
+                       "{YYYYMMDD}T{HHMM}Z "
+                       " at which to interpolate "
+                       "between the two input cubes."
+                       "Where {YYYYMMDD} is year, month day "
+                       "and {HHMM} is hour and minutes e.g "
+                       "20180116T0100Z. More than one time"
+                       "can be provided separated by a space "
+                       "but if times are set interval_in_mins "
+                       "can not be set")
+
+    parser.add_argument("--output_files", metavar="OUTPUT_FILES",
+                        required=True, nargs="+",
+                        help="List of output files.")
+
+    args = parser.parse_args()
+
+    cube_t0 = load_cube(args.data_t0)
+    cube_t1 = load_cube(args.data_t1)
```

Comment:
It would add resilience to test for time here and reorder the cubes if needed.

Reply:
Done, The resulting interpolation cubes will still go from earliest to oldest